spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/RibbonCommandContext.java

```diff
@@ -124,9 +126,18 @@ public Boolean getRetryable() {
 		return params;
 	}
 
-	public InputStream getRequestEntity() {
-		return requestEntity;
-	}
+    public InputStream getRequestEntity() {
+        try {
+            if (!(requestEntity instanceof ResettableServletInputStreamWrapper)) {
+                requestEntity = new ResettableServletInputStreamWrapper(
+                        ByteStreams.toByteArray(requestEntity));
```

Comment:
Can be a problem, true. But again, enabling ribbon.OkToRetryAllOperations should be an active decision